pkg/ebpf/generic_map.go

```diff
@@ -214,21 +209,23 @@ func (g *GenericMap[K, V]) valueTypeCanUseUnsafePointer() bool {
 }
 
 // IterateWithOptions returns an iterator for the map, which transparently chooses between batch and single item
-// iterations. This version allows choosing options
-func (g *GenericMap[K, V]) IterateWithOptions(itops IteratorOptions) GenericMapIterator[K, V] {
-	if itops.BatchSize == 0 {
-		itops.BatchSize = defaultBatchSize // Default value for batch sizes. Possibly needs more testing to find an optimal default
+// iterations. This version allows choosing the batch size. Setting the batch size to 1 will force the use of the
+// single item iterator
+func (g *GenericMap[K, V]) IterateWithOptions(batchSize int) GenericMapIterator[K, V] {
```

Comment:
maybe consider renaming to `IterateWithBatchSize`? Otherwise what the number represents when glancing at the code is unclear. Otherwise we can go back to having an options struct with a `BatchSize` field.